IdentityCore/src/workplacejoin/ios/MSIDWorkPlaceJoinUtil.m

```diff
@@ -208,9 +218,15 @@ + (nullable NSString *)getWPJStringDataForIdentifier:(nonnull NSString *)identif
         MSID_LOG_WITH_CTX(MSIDLogLevelError, context, @"Encountered an error when reading teamID from keychain.");
         return nil;
     }
-    NSString *sharedAccessGroup = [NSString stringWithFormat:@"%@.com.microsoft.workplacejoin", teamId];
+    
+    if (tenantId) {
```

Comment:
Registration for this tenant can also live in v1 keychain

Reply:
If registration details for this tenant are in "v1" keychain then getWPJKeysWithTenantId method puts "v1" instead of "v2" in keychain version.
Inside getRegisteredDeviceMetadataInformation, for "v1" getWPJStringDataForIdentifier is called which puts tenantId as nil when calling getWPJStringDataFromV2ForTenantId. So, tenantId is nil at this point and it picks "v1", queries v1 keychain and gets the registration details.